core/arch/arch.c

```diff
@@ -3131,8 +3145,8 @@ check_syscall_method(dcontext_t *dcontext, instr_t *instr)
         }
     }
 #else
-    /* we assume only single method; else need multiple do_syscalls */
-    ASSERT(new_method == get_syscall_method());
+    /* There is usually a single method. */
+    ASSERT_CURIOSITY(new_method == get_syscall_method());
```

Comment:
Since we handle it maybe there's no reason to print out a curiosity.  But we still only handle combining syscall or sysenter with int 2e: we won't handle both syscall and sysenter (seems really unlikely of course), so maybe this should be a check that either the new or the old is int 2e, right?

Reply:
ok